src/main.rs

```diff
@@ -176,28 +170,26 @@ fn main() {
         let opt = opt.clone();
         thread::spawn(move || {
             let mut _start = Instant::now();
-            let mut v: Vec<Option<Hashrate>> = vec![None; opt.threads];
-            for hash in hash_recv {
-                if opt.hashrate > 0 {
-                    let duration: Duration = _start.elapsed();
-                    v[hash.worker_id as usize] = Some(hash.borrow().clone());
-                    if ! v.contains(&None) && duration.as_secs_f32() > 30.0 {
-                        let mut total: f32 = 0.0;
-                        for h in v.iter() {
-                            let h2 = h.as_ref().unwrap();
-                            if opt.hashrate > 1 {
-                                println!("{}", h2);
-                            }
-                            total += h2.value();
-                        }
-                        let (total, unit) = hashrate::get_unit(total);
-                        println!("{} = {} {} ({})",
-                                    "Total Hashrate".blue(),
-                                    format!("{:.3}", total).red(),
-                                    unit,
-                                    format!("{} Workers", v.len()).yellow());
-                        _start = Instant::now();
+            let mut v = vec![Hashrate::default(); opt.threads];
+            for (worker_id, hash) in hash_recv {
+                let duration: Duration = _start.elapsed();
+                v[worker_id as usize] = hash;
+                if duration.as_secs_f32() > 30.0 {
+                    for h in v.iter() {
+                        println!(
+                            "Worker {} Hashrate = {}",
+                            format!("#{}", worker_id).yellow(),
+                            h
+                        );
```

Comment:
Since you removed `hashrate` option from `struct Opt`, if a user has a lot of CPU, he will be spammed by the hash rate reports for each worker. You may want to consider add `hashrate` option to CLI arguments or convert this line to `log::info`.

In addition, users had reported full of `Solution Found!` flood but not having `Got new puzzle` recently. And they had need to restart uzi-miner to work properly again. Perhaps worker hash-rate might help us to identify the bug in next testnet phase.

Reply:
I thought results are written every 30s. Am i wrong? What was the purpose of hashrate flag? didn't understand

Reply:
Yes, You are right. But if a user has i.e. 48 CPU/threads, he will see 48 lines of `Worker #n Hashrate = {}` for each thread and a `Total Hashrate = {}` line for each `30 secs`. This will fill the screen with `Worker #n Hashrate = {}` messages for each `30 secs`.

Reply:
Do you think it's necessary to print out hashrate of each worker? I think the total hashrate is enough